examples/terraform/druid-migration/zookeeper/main.tf

```diff
@@ -0,0 +1,44 @@
+resource "google_storage_bucket_object" "zoo-init-script" {
+  name    = "zookeeper_script.sh"
+  content = "${data.template_file.zookeeper_config.rendered}"
+  bucket  = "${var.bucket_name}"
+}
+
+resource "google_compute_instance" "zookeeper" {
+  name         = "${var.zookeeper_name}-${count.index + 1}"
+  machine_type = "${var.zookeeper_machine_type}"
+  tags         = ["server", "zookeeper", "myid-${count.index + 1}"]
+  zone         = "${var.zone}"
+  count        = 3
+
+  boot_disk {
+    initialize_params {
+      image = "${var.zookeeper_disk}"
+    }
+  }
+
+  network_interface {
+    network       = "${var.network}"
+    access_config = {}
+  }
+
+  metadata {
+    startup-script = "echo ${count.index + 1} > /usr/share/zookeeper.id; gsutil cp gs://${var.bucket_name}/zookeeper_script.sh /opt; chmod +x /opt/zookeeper_script.sh; /opt/./zookeeper_script.sh"
```

Comment:
Startup scripts can natively be pulled from GCS: https://cloud.google.com/compute/docs/startupscript#cloud-storage